docs/database-structure.md

```diff
@@ -0,0 +1,336 @@
+- id
+- createdAt
+- updatedAt
+- objectId: The object that requires the match.
+- userId: The user that is matched.
+- matchId: The matched that is made to another user.
+- confirmed: The same as in group_user it's a way to keep track of invitations that had not been confirmed.
+- bet: boolean; For the optional exercises that require a bet. A bet can be true if the exercises is expected to pass and false otherwise.
+- result: boolean; True for correct bet.
+- path
+- campus
+- eventId: The event the object in inserted in. (Piscine, Raid, etc).
+
+### object
+
+Objects are generic representations of elements in the structure of the curriculum (Onboarding, campus, piscine, raids, etc). They are arranged in a hierachical structure to allow unlimited nesting (see [object_child](#object_child)). They also keep the timeline of the events relaying in durations (ex. `duration`, `eventDuration`, etc) specified by [attributes](attributes.md) (`attrs`).
+
+Columns:
+- id
+- name
+- type: See [object_type](#object_type) for the possible types.
+- status: (Currently not in used, but it should always be `online` to avoid errors TODO: remove at some point).
+- attrs: See [object attributes](attributes.md).
+- childrenAttrs: Attributes that are applied to all the children objects.
+- createdAt
+- updatedAt
+- externalRelationUrl (Maybe not needed).
+- authorId (Maybe not needed).
+- campus: The campus the object is related to.
+- referenceId: A reference to the base object.
+- referencedAt: The timestamp of the creation of the copy object.
+
+### object_child
+
+It keeps a child-parent relationship between two objects. It's used to encapsulate objects inside each other.
+
+Columns:
+- id
+- parentId
+- childId
+- attrs
+- key: When generating the JS object this field will be the `key` of the child.
+- index: Defines the position of the child object inside the parent object.
+
+### object_status
+
+Columns:
+- status
+Possible values:
+  - `online`
+  - `offline`
+  - `draft`
+
+### object_type
+
+Columns:
+- type
+Possible values:
+  - `onboarding`
+  - `campus`
+  - `exercise`
+  - `quest`
+  - `signup`
+  - `exam`
+  - `raid`
+  - `project`
+  - `piscine`
+
+### progress
+
+It is a register of the steps in the progression of students doing the exercises and projects.
+
+Columns:
+- id
+- createdAt
+- updateAt
+- userId
+- groupId
+- eventId
+- version
+- grade
+- isDone
+- path
+- campus
+- objectId
+
+### record
+
+Keeps a register for the bans. It can also keep the record of other situations, this is done by setting the banEndAt to a date already passed and the reason for the record.
+
+Columns:
+- id
+- userId
+- authorId
+- message
+- banEndAt
+- createdAt
+
+### registration
+
+Registrations to events (For example, an exam).
+
+Columns:
+- id
+- createdAt, When the registration was created.
+- startAt, When users can start registering to an event.
+- endAt, When the registration ends.
+- eventStartAt, When the event the registration refers to starts.
```

Comment:
> When the event the registration refers to starts.
I don't find this very clear, the event doesn't exist before this time is reach, this is the *time when the event will be started (and as such, created)*

because unlike registrations, events exists only when they start

Reply:
Ok, got it.